docs/references/phpdoc/tags/param.rst

```diff
@@ -1,57 +1,53 @@
 @param
 ======
 
-The @param tag is used to document a single argument of a function or method.
+The ``@param`` tag is used to document a single `argument of a function or method`_.
 
 Syntax
 ------
 
+.. code-block::
+
     @param [Type] [name] [<description>]
 
 Description
 -----------
 
-With the @param tag it is possible to document the type and function of a
-single argument of a function or method. When provided it MUST contain a
-Type to indicate what is expected; the description on the other hand is
-OPTIONAL yet RECOMMENDED in case of complicated structures, such as associative
-arrays.
+With the ``@param`` tag it is possible to document the :doc:`Type <../types>`
+and function of a single `argument of a function or method`_. When provided it
+MUST contain a :doc:`Type <../types>` to indicate what is expected.
+The description is OPTIONAL yet RECOMMENDED, for instance, in case of
+complicated structures, such as associative arrays.
 
-The @param tag MAY have a multi-line description and does not need explicit
+The ``@param`` tag MAY have a multi-line description and does not need explicit
 delimiting.
 
 It is RECOMMENDED when documenting to use this tag with every function and
```

Comment:
@mvriel what do you think? should we rephrase this?  This recommendation was from the time before typehints in php. 

I would recommend usage of a `@param` when the type is a collection, or cannot be expressed by a native typehint in php. The tag should also be used when the purpose of a parameter is unclear, like boolean values, which are often featuere flags.

Reply:
Side-note: as discussed off GH, I intend to pull any significant changes we make to the tag descriptions to PSR-19, so if we can come up with a better description, let's :+1: 

Also see: https://twitter.com/ashnazg/status/1273572427935756290

Reply:
I agree that we can relax the recommendation; though I would focus on the description side of things as the main function is shifting more and more towards documenting the argument's function rather than its type. I believe we can win so much more in phpDocumentor itself as well if we were to document our own elements better

Reply:
I agree with the relaxing:

`At a minimum, it is RECOMMENDED to use this tag with each argument whose code signature does not provide Type information.`

Giving a Description has always been at the discretion of the developer, and the Name is only needed to remove all ambiguity over which argument belongs to that param tag.  But Type is important, and if it's not in the code signature, we should stress the importance of including it in the param tag.